corehq/apps/sso/views/accounting_admin.py

```diff
@@ -0,0 +1,226 @@
+from memoized import memoized
+
+from django.contrib import messages
+from django.core.exceptions import ObjectDoesNotExist
+from django.utils.decorators import method_decorator
+from django.http import Http404, HttpResponseRedirect
+from django.urls import reverse
+from django.utils.translation import ugettext as _, ugettext_lazy
+
+from corehq.apps.accounting.dispatcher import AccountingAdminInterfaceDispatcher
+from corehq.apps.accounting.filters import (
+    DateCreatedFilter,
+    NameFilter,
+)
+from corehq.apps.accounting.interface import AddItemInterface
+from corehq.apps.accounting.views import AccountingSectionView
+from corehq.apps.hqwebapp.async_handler import AsyncHandlerMixin
+from corehq.apps.reports.datatables import DataTablesHeader, DataTablesColumn
+from corehq.apps.sso.certificates import get_certificate_response
+from corehq.toggles import ENTERPRISE_SSO
+
+from corehq.apps.sso.forms import (
+    CreateIdentityProviderForm,
+    EditIdentityProviderAdminForm,
+)
+from corehq.apps.sso.async_handlers import (
+    Select2IdentityProviderHandler,
+    IdentityProviderAdminAsyncHandler,
+    SSOExemptUsersAdminAsyncHandler,
+)
+from corehq.apps.sso.models import IdentityProvider
+
+
+class IdentityProviderInterface(AddItemInterface):
+    section_name = 'Accounting'
+    name = 'Identity Providers (SSO)'
+    description = 'For managing identity providers for Single Sign On (SSO)'
+    slug = 'identity_providers'
+    dispatcher = AccountingAdminInterfaceDispatcher
+    hide_filters = False
+    item_name = "Identity Provider"
+
+    fields = [
+        'corehq.apps.accounting.interface.DateCreatedFilter',
+        'corehq.apps.accounting.interface.NameFilter',
+    ]
+
+    @classmethod
+    def show_in_navigation(cls, domain=None, project=None, user=None):
+        return ENTERPRISE_SSO.enabled(user.username)
+
+    @property
+    def new_item_view(self):
+        return NewIdentityProviderAdminView
+
+    @property
+    def headers(self):
+        return DataTablesHeader(
+            DataTablesColumn("Name"),
+            DataTablesColumn("Slug"),
+            DataTablesColumn("Edit Status"),
+            DataTablesColumn("Active Status"),
+            DataTablesColumn("Account Owner Name"),
+        )
+
+    @property
+    def rows(self):
+        def _idp_to_row(idp):
+            edit_url = reverse(EditIdentityProviderAdminView.urlname, args=(idp.id,))
+            return [
+                f'<a href="{edit_url}">{idp.name}</a>',
```

Comment:
This is an XSS vulnerability. The others were just good practice, this code is actually vulnerable to it.  Whenever you are working with HTML fragments, use `format_html()`

Reply:
the idp name is always *only* available for accounting admins to edit, but yes I'll make sure to swap this to `format_html` in a followup PR

Reply:
addressed in https://github.com/dimagi/commcare-hq/pull/29205